### PR TITLE
Fix duplicate System logging issue (#747)

### PR DESCRIPTION
**Pull request by mathewc**
Fix for https://github.com/Azure/azure-webjobs-sdk-script/issues/747

### CODE REVIEW

File: src/WebJobs.Script/Config/ScriptHostConfiguration.cs [hunk 1]
```diff
@@ -82,5 +83,11 @@ public ScriptHostConfiguration()
         /// there is no timeout duration.
         /// </summary>
         public TimeSpan? FunctionTimeout { get; set; }
+
+        /// <summary>
+        /// Gets or sets the restart interval to use between restart attempts
+        /// when the host is in an error state. The default is 5 seconds.
+        /// </summary>
+        public TimeSpan RestartInterval { get; set; }
```
Comment (mathewc) [inline src/WebJobs.Script/Config/ScriptHostConfiguration.cs]: For testability - enabled the regression test I added

File: test/WebJobs.Script.Tests/WebScriptHostManagerTests.cs [hunk 2]
```diff
@@ -124,6 +125,48 @@ public void SecretFilesArePurgedOnStartup()
...
+                return count > 3;
+            });
+
+            hostManager.Stop();
+            Assert.False(hostManager.IsRunning);
+
+            // regression test: previously on multiple restarts we were recomposing
+            // the writer on each restart, resulting in a nested chain of writers
+            // increasing on each restart
+            Assert.Equal(typeof(SystemTraceWriter), config.TraceWriter.GetType());
```
Comment (mathewc) [inline test/WebJobs.Script.Tests/WebScriptHostManagerTests.cs]: Before the fix, config.TraceWriter ended up being a Russian nesting doll of writers composed with writers.

File: src/WebJobs.Script.WebHost/WebScriptHostManager.cs [hunk 3]
```diff
@@ -331,21 +348,15 @@ protected override void OnInitializeConfig(ScriptHostConfiguration config)
...
+
             // Add our WebHost specific services
             var hostConfig = config.HostConfig;
             hostConfig.AddService<IMetricsLogger>(_metricsLogger);
 
-            var systemEventGenerator = hostConfig.GetService<IEventGenerator>() ?? new EventGenerator();
-            var systemTraceWriter = new SystemTraceWriter(systemEventGenerator, TraceLevel.Verbose);
-            if (config.TraceWriter != null)
-            {
-                config.TraceWriter = new CompositeTraceWriter(new TraceWriter[] { config.TraceWriter, systemTraceWriter });
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/WebScriptHostManager.cs]: Each restart, this is where we would keep recomposing over and over

File: src/WebJobs.Script/Config/ScriptHostConfiguration.cs (hunk 1 above)
Comment (fabiocav) [inline src/WebJobs.Script/Config/ScriptHostConfiguration.cs]: 👍

**Comment by fabiocav** [comment]
:shipit: !

**Pull request merged**